tests/system/Commands/Translation/LocalizationSyncTest.php

```diff
@@ -0,0 +1,214 @@
+<?php
+
+declare(strict_types=1);
+
+/**
+ * This file is part of CodeIgniter 4 framework.
+ *
+ * (c) CodeIgniter Foundation <admin@codeigniter.com>
+ *
+ * For the full copyright and license information, please view
+ * the LICENSE file that was distributed with this source code.
+ */
+
+namespace CodeIgniter\Commands\Translation;
+
+use CodeIgniter\Test\CIUnitTestCase;
+use CodeIgniter\Test\StreamFilterTrait;
+use Config\App;
+use Locale;
+use PHPUnit\Framework\Attributes\Group;
+
+/**
+ * @internal
+ */
+#[Group('Others')]
+final class LocalizationSyncTest extends CIUnitTestCase
+{
+    use StreamFilterTrait;
+
+    private static string $locale;
+    private static string $languageTestPath;
+
+    /**
+     * @var array<string, array<string,mixed>|string|null>
+     */
+    private array $expectedKeys = [
+        'a' => 'Sync.a',
+        'b' => 'Sync.b',
+        'c' => 'Sync.c',
+        'd' => [],
+        'e' => 'Sync.e',
+        'f' => [
+            'g' => 'Sync.f.g',
+            'h' => [
+                'i' => 'Sync.f.h.i',
+            ],
+        ],
+    ];
+
+    protected function setUp(): void
+    {
+        parent::setUp();
+
+        config(App::class)->supportedLocales = ['en', 'ru', 'test'];
+
+        self::$locale           = Locale::getDefault();
+        self::$languageTestPath = SUPPORTPATH . 'Language' . DIRECTORY_SEPARATOR;
+        $this->makeLanguageFiles();
+    }
+
+    protected function tearDown(): void
+    {
+        parent::tearDown();
+
+        $this->clearGeneratedFiles();
+    }
+
+    public function testSyncDefaultLocale(): void
+    {
+        command('lang:sync --target test');
+
+        $langFile = self::$languageTestPath . 'test/Sync.php';
+
+        $this->assertFileExists($langFile);
+
+        $langKeys = include $langFile;
+
+        $this->assertIsArray($langKeys);
+        $this->assertSame($this->expectedKeys, $langKeys);
+    }
+
+    public function testSyncWithLocaleOption(): void
+    {
+        command('lang:sync --locale ru --target test');
+
+        $langFile = self::$languageTestPath . 'test/Sync.php';
+
+        $this->assertFileExists($langFile);
+
+        $langKeys = include $langFile;
+
+        $this->assertIsArray($langKeys);
+        $this->assertSame($this->expectedKeys, $langKeys);
+    }
+
+    public function testSyncWithExistTranslation(): void
+    {
+        // First run, add new keys
+        command('lang:sync --target test');
+
+        $langFile = self::$languageTestPath . 'test/Sync.php';
+
+        $this->assertFileExists($langFile);
+
+        $langKeys = include $langFile;
+
+        $this->assertIsArray($langKeys);
+        $this->assertSame($this->expectedKeys, $langKeys);
+
+        // Second run, save old keys
+        $oldLangKeys = [
+            'a' => 'old value 1',
+            'b' => 2000,
+            'c' => null,
```

Comment:
I think we do not accept `int` or `null` as lang messages.
So should show an error message when processing?

Reply:
Maybe. These values are more for the translation search test.
I can filter out the bad lines for RU keys, if it's important.

Reply:
Please do one of the following:
- show errors (or just throw Exception?) if there are invalid values in any lang files.
- add comments in the test code that "values other than `string` is invalid, but we use invalid values just for testing".